src/link_manager/LinkManagerStateMachine.cpp

```diff
@@ -974,10 +974,17 @@ void LinkManagerStateMachine::startMuxWaitTimer(uint32_t factor)
 void LinkManagerStateMachine::handleMuxWaitTimeout(boost::system::error_code errorCode)
 {
     if (errorCode == boost::system::errc::success) {
+        mMuxWaitTimeoutCount++;
         if (mMuxStateMachine.getWaitStateCause() == mux_state::WaitState::WaitStateCause::SwssUpdate) {
             MUXLOGTIMEOUT(mMuxPortConfig.getPortName(), "orchagent timed out responding to linkmgrd", mCompositeState);
         } else if (mMuxStateMachine.getWaitStateCause() == mux_state::WaitState::WaitStateCause::DriverUpdate) {
             MUXLOGTIMEOUT(mMuxPortConfig.getPortName(), "xcvrd timed out responding to linkmgrd", mCompositeState);
+            // on the 3rd timeout, send switch active command to peer
+            if (mMuxWaitTimeoutCount == mMuxPortConfig.getNegativeStateChangeRetryCount()) { 
```

Comment:
Should the condition be >=?

Reply:
> Should the condition be >=?

In Line 991 the mux wait timer will be started again, basically linkmgrd will keep restarting the timer when it times out, and keep printing log message. So we don't want to repeat this action in later timeouts.